src/include/sof/trace.h

```diff
@@ -105,79 +107,151 @@
 #define TRACE_CLASS_IDC		(24 << 24)
 #define TRACE_CLASS_CPU		(25 << 24)
 
-/* trace core id */
-#define TRACE_CORE_ID(x)	((uint64_t)(x) << 32)
-
 /* move to config.h */
 #define TRACE	1
 #define TRACEV	0
 #define TRACEE	1
 #define TRACEM	0 /* send all trace messages to mbox and local trace buffer */
 
-void _trace_event(uint32_t event);
-void _trace_event_mbox(uint32_t event);
-void _trace_event_atomic(uint32_t event);
-void _trace_event_mbox_atomic(uint32_t event);
+void _trace_event0(uint32_t log_entry);
+void _trace_event_mbox0(uint32_t log_entry);
+void _trace_event_atomic0(uint32_t log_entry);
+void _trace_event_mbox_atomic0(uint32_t log_entry);
+
+void _trace_event1(uint32_t log_entry, uint32_t param);
+void _trace_event_mbox1(uint32_t log_entry, uint32_t param);
+void _trace_event_atomic1(uint32_t log_entry, uint32_t param);
+void _trace_event_mbox_atomic1(uint32_t log_entry, uint32_t param);
+
+void _trace_event2(uint32_t log_entry, uint32_t param1, uint32_t param2);
+void _trace_event_mbox2(uint32_t log_entry, uint32_t param1, uint32_t param2);
+void _trace_event_atomic2(uint32_t log_entry, uint32_t param1, uint32_t param2);
+void _trace_event_mbox_atomic2(uint32_t log_entry, uint32_t param1,
+	uint32_t param2);
+
+void _trace_event3(uint32_t log_entry, uint32_t param1, uint32_t param2,
+	uint32_t param3);
+void _trace_event_mbox3(uint32_t log_entry, uint32_t param1, uint32_t param2,
+	uint32_t param3);
+void _trace_event_atomic3(uint32_t log_entry, uint32_t param1, uint32_t param2,
+	uint32_t param3);
+void _trace_event_mbox_atomic3(uint32_t log_entry, uint32_t param1,
+	uint32_t param2, uint32_t param3);
+
 void trace_flush(void);
 void trace_off(void);
 void trace_init(struct sof *sof);
 
+#define PP_NARG(...) (sizeof((unsigned int[]){0, ##__VA_ARGS__}) \
+	/ sizeof(unsigned int) - 1)
+
 #if TRACE
 
+#define _DECLARE_LOG_ENTRY(lvl, format, comp_id, params)\
+	__attribute__((section(".static_log." #lvl)))	\
+	__attribute__((aligned(128)))			\
+	static const struct {				\
```

Comment:
ok, now I see - each structure is anon variable size and placed by the linker in a memory region.  Probably best to comment this above the macro. Btw, it looks like the structure can be better aligned and compressed. There are some variables that could be uint16_t and all int32_t types should be aligned on 32bits (one is placed after a variable size char array).

Reply:
Since these are never actually processed/loaded by the DSP, and are only read by external tools, I believe that we should optimize for ease of processing, rather than squeeze every bit possible.